spikesorters/yass/yass.py

```diff
@@ -0,0 +1,236 @@
+        
+        recording.write_to_binary_dat_format(input_file_path, 
+                                             dtype='int16', 
+                                             chunk_mb=500) # time_axis=0,1 for C/F order
+        
+        #################################################################
+        #################### SAVE UPDATED CONFIG FILE ###################
+        #################################################################
+        fname_config = os.path.join(output_folder,
+                                   'config.yaml')
+        with open(fname_config, 'w') as file:
+            documents = yaml.dump(self.yass_params, file)
+        
+        #################################################################
+        ############ RUN NN TRAINING ON EXISTING DATASET ################
+        #################################################################
+        self.train(recording, output_folder)
+        
+        #################################################################
+        ############ RUN NN TRAINING ON EXISTING DATASET ################
+        #################################################################
+        self.NNs_update_location(output_folder)
+
+
+        #self.fname_config = fname_config
+        
+        # ALESSIO:
+        # Expose more config file parameters that are sensiitive:
+        #  e.g. spike width; smallest cluster; min firing rates;
+        #  
+
+    def _run(self, recording, output_folder):
+        '''
+        '''
+        recording = recover_recording(recording)  # allows this to run on multiple jobs (not just multi-core)
+        
+        if 'win' in sys.platform and sys.platform != 'darwin':
+            shell_cmd = '''
+                        yass sort {config}
+                    '''.format(config=os.path.join(output_folder,'config.yaml'))
+        else:
+            shell_cmd = '''
+                        #!/bin/bash
+                        yass sort {config}
+                    '''.format(config=os.path.join(output_folder,'config.yaml'))
+
+        shell_script = ShellScript(shell_cmd, 
+                                   script_path=os.path.join(output_folder,self.sorter_name),
+                                   log_path=os.path.join(output_folder,self.sorter_name+'.log'), 
+                                   verbose=self.verbose)
+        shell_script.start()
+
+        retcode = shell_script.wait()
+
+        if retcode != 0:
+            raise Exception('yass returned a non-zero exit code')
+
+    # Alessio might not want to put here; 
+    # better option to have a parameter "tune_nn" which 
+    def train(self, recording, output_folder):
+        ''' Train NNs on yass prior to running yass sort
+        '''
+        print ("TRAINING YASS (Note: using default spike width, radius etc.)")
+        print ("To use previously-trained NNs, change the NNs prior to running: ")
+        print ("            ss.set_NNs('path_to_NNs')")
```

Comment:
I'd rather add the neural net path as a parameter

Reply:
Yes, this is done now.